resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/ReaderProvider.java

```diff
@@ -27,6 +27,8 @@
 @Provider
 @Produces("*/*")
 @Consumes("*/*")
+// FIXME: this does not implement AsyncMessageBodyWriter, but I think the current implementation is flawed as we read chars and write
+// them as if they were bytes, which will never work outside of ASCII
```

Comment:
yea, this looks really broken on the write side. I don't think it would be common to return a Reader from a resource method, which is likely why nobody has noticed.